tapdb/assets_store.go

```diff
@@ -2702,6 +2739,70 @@ func (a *AssetStore) QueryProofTransferLog(ctx context.Context,
 	return timestamps, err
 }
 
+// ConfirmProofDelivery marks a transfer output proof as successfully
+// delivered to counterparty.
+func (a *AssetStore) ConfirmProofDelivery(ctx context.Context,
+	args tapfreighter.ProofDeliveryConfirmation) error {
+
+	// Prepare arguments for use in selecting the correct transfer output
+	// row in the database.
+	//
+	// Serialise witness data.
+	var (
+		witnessBuf bytes.Buffer
+		scratch    [8]byte
+	)
+	err := asset.WitnessEncoder(&witnessBuf, &args.WitnessData, &scratch)
+	if err != nil {
+		return fmt.Errorf("unable to encode witness: %w", err)
+	}
+
+	// Serialize the script key.
+	serialisedScriptKey := args.ScriptKey.PubKey.SerializeCompressed()
+
+	// Deconstruct the split commitment root if provided.
+	var (
```

Comment:
super-nit: can just these `var` blocks together and remove `splitCommitmentRootHash`.

Non-blocking.